tracks/erlang/exercises/bob/mentoring.md

```diff
@@ -0,0 +1,46 @@
+### Reasonable solutions
+
+```erl
+response(Str) ->
+  response(Str, false, false, false, false).
+
+response([], false, false, false, false) ->
+  "Fine. Be that way!";
+response([], true, false, _HasOther, false) ->
+  "Whoa, chill out!";
+response([], true, false, _HasOther, true) ->
+  "Calm down, I know what I'm doing!";
+response([], _HasUpper, _HasLower, _HasOther, false) ->
+  "Whatever.";
+response([], _HasUpper, _HasLower, _HasOther, true) ->
+  "Sure.";
+response([C|Str], HasUpper, HasLower, HasOther, MaybeQuestion) when C=<$\s ->
+  response(Str, HasUpper, HasLower, HasOther, MaybeQuestion);
+response([$?|Str], HasUpper, HasLower, HasOther, _MaybeQuestion) ->
+  response(Str, HasUpper, HasLower, HasOther, true);
+response([C|Str], _HasUpper, HasLower, HasOther, _MaybeQuestion) when C>=$A, C=<$Z ->
+  response(Str, true, HasLower, HasOther, false);
+response([C|Str], HasUpper, _HasLower, HasOther, _MaybeQuestion) when C>=$a, C=<$z ->
+  response(Str, HasUpper, true, HasOther, false);
+response([_|Str], HasUpper, HasLower, _HasOther, _MaybeQuestion) ->
+  response(Str, HasUpper, HasLower, true, false).
+```
+
+### Common suggestions
+
+* If the student is using regular expressions, ask for a solution that works
+  without them.
+* If the student is using the `string` module, ask for a solution that works
+  without it. An exception could be if the student is trying to accomodate for
+  unicode input.
+* If the student's solution requires traversing the input string multiple
+  times (to check for uppercase letters, lowercase letters, ...), ask for
+  a solution that requires only one pass.
```

Comment:
But that's not what's in the [example solution](https://github.com/exercism/erlang/blob/master/exercises/bob/src/example.erl); are you sure that Erlang mentors agree on this?

I don't know if you can say that a multiple-pass solution is worse on all accounts.

Because each individual predicate's definition is isolated into its own function, such solutions tend to be very readable.

So it may conversely just as well be a case of premature optimization to seek the "fused" solution.

I remember from taking a course in Standard ML at university that after having completed an exercise, I'd realize that I was performing a lot more traversals than I needed, after which point I sought to meld every single logical step into one big fold. The code became unreadable, but used the fewest amount of traversals. When taking the Haskell course, I realized that I could get such fusion for free without having to think about it. Since Erlang does not provide such fusion, one is left with a pragmatic choice.